test/libsolidity/smtCheckerTests/inheritance/diamond_super_3.sol

```diff
@@ -0,0 +1,35 @@
+pragma experimental SMTChecker;
+
+contract A {
+	int public x;
+	function f() public virtual {
+		x = 1;
+	}
+}
+
+contract B is A {
+	function f() public virtual override {
+		super.f();
+		x += 100;
+	}
+}
+
+contract C is B {
+	function f() public virtual override {
+		super.f();
+		x += 10;
+	}
+}
+
+contract D is B {
+}
+
+contract E is C,D {
+	function f() public override(C,B) {
+		super.f();
+		assert(x == 111); // should hold
```

Comment:
The value is correct according to remix. But what is the order of calls here: I initially thought `+= 100` would be called twice.

Reply:
Actually, I was also a little bit surprised how this works, but I also tested this with Remix and set the values accordingly.
You can see the call order, as deduced by our encoding, in the CEX for the second, failed, assertion:
`E.f() -> C.f() -> B.f() -> A.f()`.
Is there any easy way of tracking this in Remix?

Reply:
Probably using an `event`. For example https://gist.github.com/hrkrshnn/259c206f19844f92d14e9909876d1ef1.

Actually, at some point `type(super).name` was implemented. But we decided not to merge it.

Reply:
The event idea is a bit flawed, because `type(B).name` is hardcoded. But `type(this).name` doesn't get type checked.

Reply:
You can just emit an event with a number for each contract